tools/js-tools/eslintrc/base.mjs

```diff
@@ -325,6 +325,18 @@ export default function makeBaseConfig( configurl, opts = {} ) {
 					'error',
 					{ allowInterfaces: 'with-single-extends' },
 				],
+				// Disable param name check for destructured params.
+				'jsdoc/check-param-names': [ 'error', { checkDestructured: false } ],
+				// TS should be enough to catch undefined types.
+				'jsdoc/no-undefined-types': 'warn',
```

Comment:
jsdoc's recommended-typescript config turns this off entirely, why are you doing `'warn'` instead?

Note in our CI it doesn't matter whether something is 'error' or 'warn', it will still have to be fixed for the check to pass.

Reply:
Turned it off and fixed the resulting lints via https://github.com/Automattic/jetpack/pull/40584/commits/9ef971595f14c95658882834bdae6df0a365dd52

Reply:
Ah, I see, turning off `jsdoc/no-undefined-types` makes types that are only referenced in jsdoc now be considered unused by `@typescript-eslint/no-unused-vars`.

I'm fine with 9ef9715. An alternative, to effectively turn off `jsdoc/no-undefined-types` but still mark types used in jsdocs as "used", is apparently to do like this:
```
// Mark types in jsdoc as used without reporting about any that are undefined.
'jsdoc/no-undefined-types': [ 'warn', { disableReporting: true } ],
```

Reply:
Done in https://github.com/Automattic/jetpack/pull/40584/commits/c6a0ce667edf6f801b61000ebf8b3b3c8d3f42c0

Thanks